xbmc/pvr/addons/PVRClients.cpp

```diff
@@ -33,15 +33,15 @@ using namespace PVR;
 
 namespace
 {
-  int ClientIdFromAddonId(const std::string& strID)
-  {
-    std::hash<std::string> hasher;
-    int iClientId = static_cast<int>(hasher(strID));
-    if (iClientId < 0)
-      iClientId = -iClientId;
-    return iClientId;
-  }
-
+int ClientIdFromAddonId(const std::string& addonID, ADDON::AddonInstanceId instance)
+{
+  std::hash<std::string> hasher;
+  int iClientId =
+      static_cast<int>(hasher((instance > 0 ? "@" + std::to_string(instance) : "") + addonID));
```

Comment:
Shouldn't the string to hash read "instance@addonID"? Current implementation produces for non-empty instance "@instanceaddonID"...